test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -509,5 +512,102 @@ private static void WaitForFileNotExists(FileInfo file)
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, XunitLogger);
+            var nugetTestService = GetNuGetTestService();
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(packageName, installVersion);
+            VisualStudio.ClearWindows();
+            uiwindow.UpdatePackageFromUI(packageName, updateVersion);
+
+            // Assert
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, project, packageName, updateVersion, XunitLogger);
+        }
+
+        public static IEnumerable<object[]> UpdateWebSiteTemplates()
+        {
+            yield return new object[] { ProjectTemplate.WebSiteEmpty, "WebSiteEmpty", ProjectTargetFramework.V48, "log4net", "2.0.12", "2.0.15" };
+            yield return new object[] { ProjectTemplate.WebSiteWCF, "WebSiteWCF", ProjectTargetFramework.V48, "log4net.Ext.Json", "2.0.9.1", "2.0.10.1" };
+            yield return new object[] { ProjectTemplate.WebSiteDynamicDataEntityFramework, "WebSiteEntity", ProjectTargetFramework.V48, "log4net", "2.0.13", "2.0.15" };
+        }
+
+        [NuGetWpfTheory]
+        [MemberData(nameof(UninstallWebSiteTemplates))]
+        public void UninstallWebSitePackageFromUI(ProjectTemplate projectTemplate, string projectName, ProjectTargetFramework targetFramework, string packageName, string packageVersion)
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+            var dte = VisualStudio.Dte;
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution();
+
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, targetFramework, projectName);
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, XunitLogger);
+            var nugetTestService = GetNuGetTestService();
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(packageName, packageVersion);
+            VisualStudio.ClearWindows();
+            uiwindow.UninstallPackageFromUI(packageName);
+
+            // Assert
+            CommonUtility.AssertPackageNotInPackagesConfig(VisualStudio, project, packageName, XunitLogger);
+        }
+
+        public static IEnumerable<object[]> UninstallWebSiteTemplates()
+        {
+            yield return new object[] { ProjectTemplate.WebSiteEmpty, "WebSiteEmpty", ProjectTargetFramework.V48, "log4net", "2.0.15" };
+            yield return new object[] { ProjectTemplate.WebSiteWCF, "WebSiteWCF", ProjectTargetFramework.V48, "log4net.Ext.Json", "2.0.10.1" };
+            yield return new object[] { ProjectTemplate.WebSiteDynamicDataEntityFramework, "WebSiteEntity", ProjectTargetFramework.V48, "log4net", "2.0.15" };
```

Comment:
Are `WebSiteWCF` and `WebSiteDynamicDataEntityFramework` fundamentally different than `WebSiteEmpty`?

A "normal" WCF project is just an ASP.NET project with some different packages pre-installed. So, I expect that the WebSiteWCF is also just a "normal" website project with WCF packages preinstalled, and a WCF specific `.cs` file.  

Similarly, the EntityFramework template sounds like it's just a website project with EntityFramework packages installed.

If it's realistically feasible for a test to fail one on the WCF project type, while it passes on the empty Web Site project type, then we should keep it. But I think they all use the same Web Site project system, so that they will all behave the same way, and we can reduce new test duration by 2/3rds by removing redundant tests.

Reply:
Hi @zivkan, thanks for your detailed comments. There is no difference in the WebSiteWCF and WebSiteDynamicDataEntityFramework fundamentally than WebSiteEmpty. 

Sorry, we just considered to automate the manual test cases and we didn't think about the time or performance for merging the PR. Yes, you are right, we should delete the case behave the same way and reduce the duration. Maybe keeping the WebSiteEmpty cases is enough.

Reply:
In the long term, I think it's fine if we have a whole bunch of tests that run once a week or once a day, but we don't have this yet.  I think it's also valuable to have a minimal set of VS tests that run on every pull request, to minimize risk of bugs.

So, they're probably fine tests to have, but we don't have the infrastructure set up yet.

Reply:
@zivkan 
Is the suggestion to leave the test cases in for now?

Reply:
Sorry for being unclear. I'd still like to have tests de-duplicated.

I created a new issue for the daily/weekly test idea: https://github.com/NuGet/Client.Engineering/issues/2020.  Once that is complete, these tests could be a candidate for the new pipeline.  But for this pipeline that blocks pull requests from being merged, I would like us to have a balance of "catch regressions" vs "takes a long time to run", because when we have an urgent fix to insert, having to wait a long time for the official build to complete puts us at risk of missing deadlines.